src/main/java/net/imagej/ops/segment/hough/HoughTransformOpNoWeights.java

```diff
@@ -0,0 +1,116 @@
+package net.imagej.ops.segment.hough;
+
+import org.scijava.ItemIO;
+import org.scijava.Priority;
+import org.scijava.app.StatusService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.Cursor;
+import net.imglib2.Dimensions;
+import net.imglib2.FinalDimensions;
+import net.imglib2.IterableInterval;
+import net.imglib2.img.Img;
+import net.imglib2.img.ImgFactory;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+/**
+ * Hough transform for binary images.
+ * 
+ * @author Jean-Yves Tinevez
+ *
+ * @param <T>
+ *            the type of source image. Must extend boolean type.
+ */
+@Plugin( type = HoughCircleTransformOp.class, priority = Priority.HIGH )
+public class HoughTransformOpNoWeights< T extends BooleanType< T > >
+		extends AbstractUnaryHybridCF< IterableInterval< T >, Img< DoubleType > >
+		implements Contingent
+{
+
+	@Parameter
+	protected StatusService statusService;
+
+	@Parameter( label = "Min circle radius", description = "Minimal radius, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT )
+	protected long minRadius = 1;
+
+	@Parameter( label = "Max circle radius", description = "Maximal radius, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT )
+	protected long maxRadius = 50;
+
+	@Parameter( label = "Step radius", description = "Radius step, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT, required = false )
+	protected long stepRadius = 1;
+
+
+	@Override
+	public Img< DoubleType > createOutput( final IterableInterval< T > input )
+	{
+		// Get a suitable image factory.
+		final int numDimensions = input.numDimensions();
+		if ( input.numDimensions() != 2 ) { throw new IllegalArgumentException(
+				"Cannot compute Hough circle transform for non-2D images. Got " + numDimensions + "D image." ); }
+
+		maxRadius = Math.max( minRadius, maxRadius );
```

Comment:
If the user inputs the smaller value in `maxRadius` and the larger value in `minRadius` then the smaller value will be overwritten with this logic. You need a temporary value here to store the value of `maxRadius` in the case that it is overwritten, I think.

Reply:
Doh! My bad.